packages/login/src/components/TwoStepLoginForm/TwoStepLoginForm.js

```diff
@@ -1,90 +1,69 @@
+const TwoStepLoginForm = ({username, password, showTitle, loginPending, twoStepLogin, intl}) => {
+  const [userCode, setUserCode] = useState(null)
+  const formEl = useRef()
 
-  handleSubmit(e) {
-    e.preventDefault()
-    this.props.twoStepLogin(this.props.username, this.props.password, this.props.requestedCode, this.state.userCode)
-  }
+  useLayoutEffect(() => {
+    const firstInput = formEl.current.querySelector('input')
+    firstInput.focus()
+  }, [])
 
-  handleUserCodeChange(e) {
-    this.setState({
-      userCode: e.target.value
-    })
+  const handleSubmit = e => {
+    e.preventDefault()
+    twoStepLogin(username, password, userCode)
   }
 
-  render() {
-    return (
-      <StyledTwoStepLogin>
-        {this.props.showTitle
-          && <Typography.H5><FormattedMessage id="client.login.form.title"/></Typography.H5>}
-        <form onSubmit={this.handleSubmit.bind(this)}>
-          <Typography.P><FormattedMessage id="client.login.twoStepLogin.introduction"/></Typography.P>
-          <Typography.P>
-            <FormattedMessage id="client.login.twoStepLogin.requestedCode"/>{this.props.requestedCode}
-          </Typography.P>
-          <StatedValue
-            hasValue={!!this.state.userCode}
-            id={'usercode'}
-            label={this.msg('client.login.twoStepLogin.codePlaceholder')}
-          >
-            <StyledTwoStepLoginInputWrapper>
-              <StyledTwoStepLoginInput
-                autoComplete="off"
-                id={'usercode'}
-                value={this.state.username}
-                onChange={this.handleUserCodeChange.bind(this)}
-                name="code"
-              />
-            </StyledTwoStepLoginInputWrapper>
-          </StatedValue>
+  const msg = id => intl.formatMessage({id})
 
-          <StyledLoginButton
-            disabled={!this.state.userCode || this.props.loginPending}
-            ink="primary"
-            label={this.msg('client.login.form.button')}
-            look="raised"
-            name="submit"
-            pending={this.props.loginPending}
-            type="submit"
-          />
-        </form>
-      </StyledTwoStepLogin>
-    )
-  }
-
-  msg(id) {
-    return this.props.intl.formatMessage({
-      id
-    })
-  }
+  return <StyledTwoStepLogin>
+    {showTitle && <Typography.H5><FormattedMessage id="client.login.form.title"/></Typography.H5>}
+    <form onSubmit={handleSubmit} ref={formEl}>
+      <Typography.P><FormattedMessage id="client.login.twoStepLogin.introduction"/></Typography.P>
+      <StatedValue
+        hasValue={!!userCode}
+        id="twoFactorCode-label"
+        label={msg('client.login.twoStepLogin.codePlaceholder')}
+      >
+        <EditableValue
+          type="number"
+          value={userCode}
+          events={{onChange: setUserCode}}
+          options={{format: '### ###'}}
+        />
+      </StatedValue>
+      <StyledLoginButton
+        disabled={!userCode || userCode.toString().length < 6 || loginPending}
```

Comment:
disable if length `!== 6` instead of `< 6` ? or isn't it possible to enter more than 6 digits?

Reply:
It's not possible to enter more than 6 characters.